examples/flow.cpp

```diff
@@ -184,15 +252,17 @@ try
         ioConfig->overrideRestartWriteInterval((size_t)output_interval);
     }
 
-    Opm::BlackoilOutputWriter outputWriter(cGrid,
-                                           param,
-                                           eclipseState,
-                                           pu );
+    const PhaseUsage pu = Opm::phaseUsageFromDeck(deck);
+    Opm::BlackoilOutputWriter outputWriter(grid, param, eclipseState, pu );
 
     // Rock and fluid init
-    props.reset(new BlackoilPropertiesFromDeck(deck, eclipseState, *grid->c_grid(), param));
-    new_props.reset(new BlackoilPropsAdFromDeck(deck, eclipseState, *grid->c_grid()));
-
+    props.reset(new BlackoilPropertiesFromDeck(deck, eclipseState,
+                                               Opm::UgGridHelpers::numCells(grid),
+                                               Opm::UgGridHelpers::globalCell(grid),
+                                               Opm::UgGridHelpers::cartDims(grid),
+                                               Opm::UgGridHelpers::beginCellCentroids(grid),
+                                               Opm::UgGridHelpers::dimensions(grid), param));
+    new_props.reset(new BlackoilPropsAdFromDeck(deck, eclipseState, grid));
```

Comment:
If the objects create objects on the heap themselves thats ok, otherwise you can run into trouble with stack size limitations.